test/Unity.TypedFactories.Tests/UnityTypedFactoryTests.cs

```diff
@@ -160,7 +161,8 @@ public void given_instantiated_Sut_when_Create_is_called_and_ArgumentException_i
 
                 // Act
                 var factory = unityContainer.Resolve<ISomeInstanceFactory>();
-                factory.Create();
+                
+                Assert.Throws<System.Reflection.TargetInvocationException>(() => factory.Create());
```

Comment:
Would be interesting to unwrap the `TargetInvocationException` exception and assert on what is inside it (probably the `InvalidOperationException`?)

Reply:
I pushed a re-work of that so it will unwrap the exception.